core/src/main/java/org/jdbi/v3/core/vendor/H2DatabasePlugin.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.vendor;
+
+import java.util.UUID;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.SqlArrayConfiguration;
+import org.jdbi.v3.core.SqlArrayConfiguration.ArgumentStyle;
+import org.jdbi.v3.core.spi.JdbiPlugin;
+
+public class H2DatabasePlugin implements JdbiPlugin
+{
+    @Override
+    public void customizeDbi(Jdbi dbi)
+    {
+        dbi.configure(SqlArrayConfiguration.class, c -> c.withArgumentStyle(ArgumentStyle.OBJECT_ARRAY));
```

Comment:
We could use this configuration stuff for reflective mapper column name matching strategies too.